Topic5/src/Demo.java

```diff
@@ -0,0 +1,131 @@
+import java.sql.SQLException;
+import java.util.Scanner;
+
+public class Demo {
+
+	public static void main(String[] args) {
+		
+		java.sql.Connection connection;
+		try {
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+
+			// connection to the DB
+			String servidor = "localhost:3306";
+			String baseDatos = "high_school";
+			String uriConexion = "jdbc:mysql://" + servidor + "/" + baseDatos;
+			String user = "adminBD";
+			String password = "admin";
+
+			connection = java.sql.DriverManager.getConnection(uriConexion, user, password);
+
+			java.sql.Statement statement = connection.createStatement();
+
+			// KEY POINT 4
+			System.out.println("List students and teachers for a given course.");
+			System.out.println("Courses: ");
+			System.out.println("1.Java course 2.PHP course 3.C course 4.C++ course 5.Phyton course");
+			System.out.print("Choose a course number: ");
+
+			// the id_course chosen by the user
+			int id_course;
+
+			Scanner scanner = new Scanner(System.in);
+			id_course = scanner.nextInt();
+
+			// the id_course must be between 1 and 5
+			while (id_course < 1 || id_course > 5) {
+				System.out.println();
+				System.out.println("Wrong choice, please choose one of the following number course: ");
+				System.out.println("1.Java course 2.PHP course 3.C course 4.C++ course 5.Phyton course");
+				System.out.print("Choose a course number: ");
+				id_course = scanner.nextInt();
+			}
+
+			System.out.println();
+
+			// query to obtain the name of the course and the first_name and
+			// last_name of the teacher
+			String sql = "SELECT name_course,first_name,last_name FROM Course NATURAL JOIN Teacher WHERE id_course=" + id_course;

```

Comment:
It's a good practice to analyze in each case which kind of JOIN you need to use: In this case I would use an INNER JOIN since we need to retrieve all the records in the left table (Course) that have a matching record in the right table (Teacher), but if we are looking for all the courses independetly if they have a teacher assigned or not we should use LEFT JOIN.
In the following page you have an excellent Visual Representation of SQL JOINS -> http://www.codeproject.com/Articles/33052/Visual-Representation-of-SQL-Joins